app/javascript/mastodon/db/modifier.js

```diff
@@ -0,0 +1,86 @@
+import asyncDB from './async';
+
+const limit = 1024;
+
+function put(name, objects, callback) {
```

Comment:
Typically it's safer and less error-prone to call the callback from the transaction's`oncomplete` callback. It's guaranteed to be called when the transaction completes, so you don't have to do any extra bookkeeping yourself.